crowbar_framework/app/models/api/node.rb

```diff
@@ -142,11 +142,44 @@ def remove_pre_upgrade_role_for(name)
     # Delete existing pacemaker resources, from other node in the cluster
     def delete_pacemaker_resources
       script = "/usr/sbin/crowbar-delete-pacemaker-resources.sh"
-      out = @node.run_ssh_cmd(script)
-      unless out[:exit_code].zero?
-        save_error_state("Deleting pacemaker resources failed on node #{@node.name}.")
-        Rails.logger.error("Script location: #{script}")
-        Rails.logger.error("stdout: #{out[:stdout]}, stderr: #{out[:stderr]}")
+      ssh_status = @node.ssh_cmd(script)
+      if ssh_status[0] != 200
+        save_error_state(
+          "Executing of 'delete pacemaker resources' has failed at node #{@node.name}.\
+Script location: #{script}."
+        )
+        return false
+      end
+
+      # wait until the the script is finished
```

Comment:
A comment stating the obvious is often an indication that the code can be split into a separate method.  In this case you could replace the comment with

    def await_pacemaker_resource_deletion

and then we'd have two easy-to-understand methods which both do one simple task well, rather than a longer and harder to understand method.